### PR TITLE
Fixed reading chunked EIP8 handshake

### LINKED ISSUE #1633: Geth 1.5 can't connect to to Parity peer
As reported in gitter:
https://gist.github.com/amiller/f7d9a34b8a0d0d6691c2543503c55a54

Connection initiated by Parity to the same geth node apparently works.

### PR DESCRIPTION
**Pull request by arkpar**
Closes #1633

### CODE REVIEW

File: util/src/network/handshake.rs [hunk 1]
```diff
@@ -128,31 +128,26 @@ impl Handshake {
...
-					if let Some(data) = try!(self.connection.readable()) {
+					},
+					HandshakeState::ReadingAckEip8 => {
 						try!(self.read_ack_eip8(host.secret(), &data));
-					};
-				},
-				HandshakeState::StartSession => {
+					},
+				}
+				if self.state == HandshakeState::StartSession {
```
Comment (gavofyork) [inline util/src/network/handshake.rs]: why moved outside the `match`?

Comment (arkpar) [inline util/src/network/handshake.rs]: `connection.readable()` Is called in a loop now. There is no need to call it once we've entered `StartSession` state

Comment (gavofyork) [inline util/src/network/handshake.rs]: yes but why not place `io.clear_timer(self.connection.token).ok();` in the braces of `HandshakeState::StartSession => {},`?

Comment (arkpar) [inline util/src/network/handshake.rs]: Added missing `break` that saves the `readable` call

**Pull request merged**